File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/runtime/AmbientCacheOperation.java
package org.maplibre.nativeffi.runtime;

/** Ambient cache maintenance operation for a runtime. */
public enum AmbientCacheOperation {
  RESET_DATABASE(1),
  PACK_DATABASE(2),
  INVALIDATE(3),
  CLEAR(4);

  private final int nativeValue;

  AmbientCacheOperation(int nativeValue) {
    this.nativeValue = nativeValue;
  }

  int nativeValue() {
    return nativeValue;
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/runtime/AmbientCacheOperation.kt
package org.maplibre.nativeffi.runtime

/** Ambient cache maintenance operation for a runtime. */
public enum class AmbientCacheOperation(public val nativeValue: Int) {
  RESET_DATABASE(1),
  PACK_DATABASE(2),
  INVALIDATE(3),
  CLEAR(4),
}
